js/src/Ice/StringUtil.js

```diff
@@ -209,12 +209,40 @@ export class StringUtil {
         return hash;
     }
 
-    static toInt(s) {
-        const n = parseInt(s, 10);
-        if (isNaN(n)) {
-            throw new RangeError("conversion of `" + s + "' to int failed");
+    static parseSafeInt32(s) {
+        // Handle edge cases first
+        if (typeof s !== 'string') {
+            throw new RangeError(`conversion of '${s}' to int failed`);
+        }
+        
+        // Check for empty string
+        if (s.length === 0) {
+            throw new RangeError(`conversion of '' to int failed`);
+        }
+        
+        // Trim whitespace and check if it becomes empty
+        const trimmed = s.trim();
+        if (trimmed.length === 0) {
+            throw new RangeError(`conversion of '${s}' to int failed`);
+        }
```

Comment:
I think it is more logical to first trim, and if `s !== trimmed` throw. 
Then do the empty check in trimmed. 
Finally parse trimmed.

The current code would accept `" 10 "` as `"10"` ,  but with the proposed fix it would be rejected. `s !== trimmed`.